src/Zicht/Bundle/UrlBundle/Aliasing/Listener.php

```diff
@@ -147,76 +168,102 @@ protected function isExcluded($url)
      */
     public function onKernelRequest(Event\GetResponseEvent $event)
     {
-        if ($event->getRequestType() === HttpKernelInterface::MASTER_REQUEST) {
-            $request = $event->getRequest();
-            $publicUrl = rawurldecode($request->getRequestUri());
+        if ($event->getRequestType() !== HttpKernelInterface::MASTER_REQUEST) {
+            return;
+        }
 
-            if ($this->isExcluded($publicUrl)) {
-                // don't process urls which are marked as excluded.
-                return;
-            }
+        $request = $event->getRequest();
+        $publicUrl = rawurldecode($request->getRequestUri());
 
-            if ($this->isParamsEnabled) {
-                if (false !== ($queryMark = strpos($publicUrl, '?'))) {
-                    $originalUrl = $publicUrl;
-                    $publicUrl = substr($originalUrl, 0, $queryMark);
-                    $queryString = substr($originalUrl, $queryMark);
-                } else {
-                    $queryString = null;
-                }
+        if ($this->isExcluded($publicUrl)) {
+            // don't process urls which are marked as excluded.
+            return;
+        }
 
-                $parts = explode('/', $publicUrl);
-                $params = array();
-                while (false !== strpos(end($parts), '=')) {
-                    array_push($params, array_pop($parts));
-                }
-                if ($params) {
-                    $publicUrl = join('/', $parts);
+        $queryString = '';
+        if (false !== ($queryMark = strpos($publicUrl, '?'))) {
+            $queryString = substr($publicUrl, $queryMark);
+            $publicUrl = substr($publicUrl, 0, $queryMark);
+        }
 
-                    $parser = new UriParser();
-                    $request->query->add($parser->parseUri(join('/', array_reverse($params))));
+        if ($this->isParamsEnabled) {
+            $parts = explode('/', $publicUrl);
+            $params = array();
+            while (false !== strpos(end($parts), '=')) {
+                array_push($params, array_pop($parts));
+            }
```

Comment:
Deze while loop gaat van het einde naar het begin en de resulterende array gaat van begin naar einde.  Voelt inconsistent.  Voorbeeld:

```
$parts = ['foo=A', 'bar', 'moo=Milk'];
// resulteerd in, dus exact de andere volgorde
$params = ['moo=Milk', 'foo=A'];
```

Gaat dit misschien later problemen opleveren als er duplicate keys in zitten die elkaar overschrijven, voorbeeld:
```
$url = '/nice-url/foo=A/foo=B';
// hier maken wij (vanwege het omkeren van de volgorde) van
$url = '/page/nl/123/foo=B/foo=A'

// Wanneer de pagina wordt ingeladen...
// resulteerd dit in het omgekeerde van wat het zonder de nice url parameter
// conversie was: ['foo' => 'A'] <--> ['foo' => 'B']?
```

Reply:
Dit is bestaande code en nu niet echt het issue, dus het lijkt mij beter om daar later nog eens over na te denken